src/Evolution/Systems/GeneralizedHarmonic/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,297 @@
+ * by \f$v_{\hat{\alpha}}\f$, and implicitly sum over reapeated indices.
+ * \f$T_{\beta\hat{\beta}}\f$ transforms characteristic fields to evolved
+ * fields, while \f$\Lambda_{\hat{\beta}\hat{\alpha}}^-\f$ is a diagonal matrix
+ * with only the negative characteristic speeds, and has zeros on the diagonal
+ * for all other entries. The int and ext superscripts denote quantities on the
+ * internal and external side of the mortar. Note that Eq. (6.3) of
+ * \cite Teukolsky2015ega is not exactly what's implemented since that boundary
+ * term does not consistently treat both sides of the interface on the same
+ * footing.
+ *
+ * For the first-order generalized harmonic system the correction is:
+ *
+ * \f{align*}{
+ *   D_{g_{ab}} &= \lambda_{v^{g}}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},g}_{ab}
+ *                - \lambda_{v^{g}}^{\mathrm{int},-}
+ *                v^{\mathrm{int},g}_{ab}, \\
+ *   D_{\Pi_{ab}}
+ *              &= \frac{1}{2}\left(\lambda_{v^+}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},+}_{ab} +
+ *                \lambda_{v^-}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},-}_{ab}\right)
+ *                + \lambda_{v^g}^{\mathrm{ext},-}\gamma_2
+ *                v^{\mathrm{ext},g}_{ab}
+ *                \notag \\
+ *              &-\frac{1}{2}\left(\lambda_{v^+}^{\mathrm{int},-}
+ *                v^{\mathrm{int},+}_{ab} +
+ *                \lambda_{v^-}^{\mathrm{int},-}
+ *                v^{\mathrm{int},-}_{ab}\right)
+ *                - \lambda_{v^g}^{\mathrm{int},-}\gamma_2
+ *                v^{\mathrm{int},g}_{ab} , \\
+ *   D_{\Phi_{iab}}
+ *              &= \frac{1}{2}\left(\lambda_{v^+}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},+}_{ab}
+ *                - \lambda_{v^-}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},-}_{ab}\right)n_i^{\mathrm{ext}}
+ *                + \lambda_{v^0}^{\mathrm{ext},-}
+ *                v^{\mathrm{ext},0}_{iab}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\lambda_{v^+}^{\mathrm{int},-}
+ *                v^{\mathrm{int},+}_{ab}
+ *                - \lambda_{v^-}^{\mathrm{int},-}
+ *                v^{\mathrm{int},-}_{ab}\right)n_i^{\mathrm{int}}
+ *                - \lambda_{v^0}^{\mathrm{int},-}
+ *                v^{\mathrm{int},0}_{iab},
+ * \f}
+ *
+ * with characteristic fields
+ *
+ * \f{align*}{
+ *   v^{g}_{ab} &= g_{ab}, \\
+ *   v^{0}_{iab} &= (\delta^k_i-n^k n_i)\Phi_{kab}, \\
+ *   v^{\pm}_{ab} &= \Pi_{ab}\pm n^i\Phi_{iab} -\gamma_2 g_{ab},
+ * \f}
+ *
+ * and characteristic speeds
+ *
+ * \f{align*}{
+ *   \lambda_{v^g} =& -(1+\gamma_1)\beta^i n_i -v^i_g n_i, \\
+ *   \lambda_{v^0} =& -\beta^i n_i -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm \alpha - \beta^i n_i - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity and \f$n_i\f$ is the outward directed
+ * unit normal covector to the interface. We have also defined
+ *
```

Comment:
this is for the definition of D below and not above?

Reply:
Both

Reply:
ah, I see, there is no \lambda^+ in the definition of D above